lib/chef/client.rb

```diff
@@ -323,6 +325,22 @@ def run
     # @todo make this stuff protected or private
     #
 
+    # @api private
+    def warn_if_eol
+      require_relative "version"
+
+      # we have a yearly release and we know 15 goes EOL in 2021 so calculate off that
+      # this way we don't have to update anything or maintain a hash of EOL dates
+      base_eol_year = 2022
+      base_release = Gem::Version.new(15)
+
+      diff_from_base_release = Gem::Version.new(Chef::VERSION).segments.first - base_release.segments.first
```

Comment:
`eol_year = 2007 + Gem::Version.new(Chef::VERSION).segments.first` ?

the double magic constants of 15 expiring in 2022 and doing arithmetic around that seems more confusing to me than just having a magic base year of 2007 that makes the math work out.

Reply:
(or 2006)

Reply:
Yeah that makes way more sense to me as well. I updated the PR